examples/bri-3/src/bri/communication/communication.module.ts

```diff
@@ -1,33 +1,43 @@
 import { Module } from '@nestjs/common';
 import { CqrsModule } from '@nestjs/cqrs';
+import { LoggingModule } from '../../shared/logging/logging.module';
 import { SubjectModule } from '../identity/bpiSubjects/subjects.module';
 import { SubjectsProfile } from '../identity/bpiSubjects/subjects.profile';
+import { NatsMessagingClient } from './/messagingClients/natsMessagingClient';
 import { BpiMessageAgent } from './agents/bpiMessages.agent';
 import { BpiMessageStorageAgent } from './agents/bpiMessagesStorage.agent';
+import { MessagingAgent } from './agents/messaging.agent';
 import { MessageController } from './api/messages.controller';
 import { CreateBpiMessageCommandHandler } from './capabilities/createBpiMessage/createBpiMessageCommand.handler';
 import { DeleteBpiMessageCommandHandler } from './capabilities/deleteBpiMessage/deleteBpiMessageCommand.handler';
 import { GetBpiMessageByIdQueryHandler } from './capabilities/getBpiMessageById/getBpiMessageByIdQuery.handler';
+import { ProcessInboundMessageCommandHandler } from './capabilities/processInboundMessage/processInboundMessageCommand.handler';
 import { UpdateBpiMessageCommandHandler } from './capabilities/updateBpiMessage/updateBpiMessageCommand.handler';
 import { CommunicationProfile } from './communicaton.profile';
 
 export const CommandHandlers = [
   CreateBpiMessageCommandHandler,
   UpdateBpiMessageCommandHandler,
   DeleteBpiMessageCommandHandler,
+  ProcessInboundMessageCommandHandler,
 ];
 export const QueryHandlers = [GetBpiMessageByIdQueryHandler];
 
 @Module({
-  imports: [CqrsModule, SubjectModule],
+  imports: [CqrsModule, SubjectModule, LoggingModule],
   controllers: [MessageController],
   providers: [
     ...CommandHandlers,
     ...QueryHandlers,
     BpiMessageAgent,
     BpiMessageStorageAgent,
+    MessagingAgent,
     SubjectsProfile,
     CommunicationProfile,
+    {
+      provide: 'IMessagingClient',
+      useClass: NatsMessagingClient, // TODO: This is the place where we would inject different clients based on configuration (i.e. RabbitMQ client)
```

Comment:
this TODO seems a bit redundant since we don't have plan to do this, at least not soon, so we won't fix it soon

Reply:
Removed TODO.